### ISSUE TITLE
[Bug]: attach-mode fails for `ssh` backend Task instance 

### ISSUE THREAD

**Issue by deep-diver** [dstackai/dstack#1246]
### Steps to reproduce

0. Add `ssh` backend instance to `default-pool`

1. Open up GitHub codespaces for a clear and fresh setup

2. run `dstack config ...`

3. run `dstack init ...`

4. run `dstack run . -f ... --reuse`

5. select the instance from the `default-pool`

### Actual behaviour

```console
fast-rat-1 provisioning completed (running)
Aborted
Traceback (most recent call last):
  File "/home/codespace/.python/current/bin/dstack", line 8, in <module>
    sys.exit(main())
  File "/usr/local/python/3.10.13/lib/python3.10/site-packages/dstack/_internal/cli/main.py", line 67, in main
    args.func(args)
  File "/usr/local/python/3.10.13/lib/python3.10/site-packages/dstack/_internal/cli/commands/run.py", line 183, in _command
    if run.attach():
  File "/usr/local/python/3.10.13/lib/python3.10/site-packages/dstack/api/_public/runs.py", line 293, in attach
    self._ssh_attach.attach()
  File "/usr/local/python/3.10.13/lib/python3.10/site-packages/dstack/_internal/core/services/ssh/attach.py", line 136, in attach
    raise SSHError("Can't connect to the remote host")
dstack._internal.core.errors.SSHError: Can't connect to the remote host
```

### Expected behaviour

it should be able to connect to the provisioned instance

### dstack version

0.18.1

### Server logs

```shell
no server logs
```

**Comment by TheBits** [comment]
Hi, Chunsung!
Thanks for bug report!

Are you using `dstack server` or dstack Sky?

I [fixed it](https://github.com/dstackai/dstack/pull/1204) in the `dstack server`, and it should work in the next version.

**Comment by deep-diver** [comment]
I am using dstack Sky, but this nodes are added by @peterschmidt85 for testing purpose

**Comment by TheBits** [comment]
I'll close the issue; the fixes will be in `0.18.2rc2`.
